pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -139,8 +139,17 @@ type mergeQuerier struct {
 func (m *mergeQuerier) LabelValues(name string, matchers ...*labels.Matcher) ([]string, storage.Warnings, error) {
 	log, _ := spanlogger.New(m.ctx, "mergeQuerier.LabelValues")
 	defer log.Span.Finish()
+
+	matchedValues, filteredMatchers := filterValuesByMatchers(m.idLabelName, m.ids, matchers...)
+
 	if name == m.idLabelName {
-		return m.ids, nil, nil
+		var labelValues []string
```

Comment:
You could create the slice in with the correct capacity from the start
```suggestion
		var labelValues = make([]string,0, len(matchedValues))
```